portfolio/src/main/webapp/leave-comment/script.js

```diff
@@ -1,8 +1,56 @@
+function onSignIn(googleUser) {
+  if ($('#sign-out').length != 0) {
+    return;
```

Comment:
Why?

The reason I'm asking is that it seems to make it impossible to sign in as a different user without explicitly signing out. BTW. Can you add information to the page about who the page considers to be signed in?

Reply:
This sign-in button allows to sign in multiple times using same account and this check is made to ensure unnecessary actions won't be made. For example, I read a description of user every time they sign in, because I have to update their token.

Reply:
But doesn't the check mean "if the user is signed in, ignore (`return`) any other attempt to sign in"? If my reading is correct, signing in as a different user will not update the IdToken.

Reply:
Ok, I understand your comment now, sorry. You are right, this is big issue.
If I allow users to sign in as a different account without signing out, I'll have to update idToken every time, but the rest of actions can be easily ommited if I use visibility change instead adding new button. Second idea is to hide/remove sign-in button to force user to sign out before signing in with another account, but first one is more comfortable for end user.

Reply:
Done.